include/chaiscript/language/chaiscript_engine.hpp

```diff
@@ -204,6 +204,27 @@ namespace chaiscript
       m_engine.add(fun([this](const std::string& t_namespace_name) { import(t_namespace_name); }), "import");
     }
 
+    /// Skip BOM at the beginning of file
+    static bool skip_bom(std::ifstream &infile) {
+        size_t bytes_needed = 3;
+        std::vector<char> v(bytes_needed);
+
+        infile.read(&v[0], static_cast<std::streamsize>(bytes_needed));
+        std::string buffer_string(v.begin(), v.end());
+
+        if ((buffer_string.size() > 2)
```

Comment:
This statement will always be true, since the vector always contains `bytes_needed` elements. In addition, this should use `bytes_needed` rather than a constant.

Overall I'd just check whether the stream has reached the end of file or not.